tasks/modules.py

```diff
@@ -443,36 +154,91 @@ def for_each(
 
 
 @task
-def validate(_: Context):
+def validate(ctx: Context, base_dir='.', fix_format=False):
     """
-    Test if every module was properly added in the DEFAULT_MODULES list.
+    Lints module configuration file.
+
+    Args:
+        fix_format: If True, will fix the format of the configuration files.
     """
-    missing_modules: list[str] = []
-    default_modules_paths = {Path(p) for p in DEFAULT_MODULES}
 
-    # Find all go.mod files and make sure they are registered in DEFAULT_MODULES
-    for root, dirs, files in os.walk("."):
-        dirs[:] = [d for d in dirs if Path(root) / d not in IGNORED_MODULE_PATHS]
+    base_dir = Path(base_dir)
+    config = Configuration.from_file(base_dir)
+    default_attributes = GoModule.get_default_attributes()
+
+    # Verify format
+    with tempfile.TemporaryDirectory() as tmpdir:
+        config.base_dir = Path(tmpdir)
+        config.to_file()
+        config.base_dir = base_dir
+
+        if not ctx.run(
+            f'diff -u {base_dir / Configuration.FILE_NAME} {Path(tmpdir) / Configuration.FILE_NAME}',
+            warn=True,
+        ):
+            if fix_format:
+                print(f'{color_message("Info", Color.BLUE)}: Formatted module configuration file')
+                config.to_file()
+            else:
+                raise Exit(
+                    f'{color_message("Error", Color.RED)}: Configuration file is not formatted correctly, use `invoke modules.validate --fix-format` to fix it'
+                )
 
-        if "go.mod" in files and Path(root) not in default_modules_paths:
-            missing_modules.append(root)
+    def validate_module(module: GoModule, attributes: str | dict[str, object]):
```

Comment:
🥜 I would probably suggest not to use inner functions except if you need it for a special purpose (closure factory or decorator) or if they are very small, as I think it decreases the readability (or testability) of code.
This is maybe a personal feedback, happy to discuss it.